### DISCUSSION TITLE
Is the "base" option from the Astro configuration supported?

### DISCUSSION THREAD

**Discussion by rumenpetrov** [jlarmstrongiv/astro-i18n-aut#17] [Q&A] [answered]
I don't see anything in the documentation about this but I was wandering is the [base](https://docs.astro.build/en/reference/configuration-reference/#base) option covered in this library because I am having problems making my application work with it?

Here are some details about my app:
- It's static
- I am using github to host it, this is why I need the base option
- I am also using lit and and prefetch

Here is how my config looks like
```js
import { defineConfig } from 'astro/config';
import { i18n, defaultLocaleSitemapFilter } from 'astro-i18n-aut/integration';
import lit from '@astrojs/lit';
import prefetch from '@astrojs/prefetch';

const defaultLocale = 'bg';
const locales = {
  bg: 'bg-BG',
  en: 'en-US',
};

// https://astro.build/config
export default defineConfig({
  site: 'https://rumenpetrov.github.io',
  output: 'static',
  trailingSlash: 'always',
  build: {
    format: 'directory',
  },
  base: '/holiday-planner',
  integrations: [
    lit(),
    prefetch(),
    i18n({
      locales,
      // defaultLocale and redirectDefaultLocale are in middleware.js as well
      defaultLocale,
      redirectDefaultLocale: true,
    }),
  ],
});
```

**Comment by jlarmstrongiv** [comment, marked as answer]
If you follow Astro convention, you should only have `.astro` pages in the `pages` directory, which means that all of the import paths for images, content, and components should stay the same.  I was careful to:
- [support](https://github.com/jlarmstrongiv/astro-i18n-aut/blob/main/src/integration/integration.ts#L44) the [`srcDir`](https://docs.astro.build/en/reference/configuration-reference/#srcdir) option
- [support](https://github.com/jlarmstrongiv/astro-i18n-aut/blob/main/src/integration/integration.ts#L119) the [`base`](https://docs.astro.build/en/reference/configuration-reference/#base) option

But, there may be a bug with the implementation.  Do you have an example on GitHub I could take a look at?  Maybe `injectPages` already prepends the base, so I don’t need to add it there.

Or did you get it to work?  I see https://github.com/rumenpetrov/holiday-planner hosted on https://rumenpetrov.github.io/holiday-planner/en/ seems to work fine.

Astro does require you to prepend your links and urls manually:
```html
<a href="/docs/about/">About</a>
<img src=`${import.meta.env.BASE_URL}image.png`>
```

**Reply by jlarmstrongiv** [reply]
@rumenpetrov following up, you may be running into https://github.com/jlarmstrongiv/astro-i18n-aut/issues/21

**Reply by rumenpetrov** [reply]
@jlarmstrongiv thank you for the feedback I assumed that I was doing something wrong, this is why I wanted to confirm that the library supports the base option. 
At the end I decided to pick a different library because I had time constraints but I could give it another try in future. I really liked the approach of this library.

**Reply by jlarmstrongiv** [reply]
@rumenpetrov base path support has been fixed in `0.3.5` 👍  it should work the next time you try it
